Modsenfy.DataAccessLayer/Repositories/ArtistRepository.cs

```diff
@@ -21,40 +14,69 @@ public ArtistRepository(DatabaseContext databaseContext)
         }
 
-        public async Task Create(Artist entity)
+        public async Task<Artist> GetById(int id)
         {
-            await _databaseContext.AddAsync(entity);
+            var artist = await _databaseContext.Artists.FindAsync(id);
+
+            return artist;
         }
 
-        public Task<IEnumerable<Artist>> GetAll()
+        public async Task<Artist> GetByIdWithJoins(int id)
         {
-            throw new NotImplementedException();
+            var artist = await _databaseContext.Artists
+                .Include(a => a.Image)
+                    .ThenInclude(i => i.ImageType)
+                .Include(a => a.TrackArtists)
+                    .ThenInclude(ta => ta.Track)
+                        .ThenInclude(t => t.Audio)
+                .Include(a => a.TrackArtists)
+                    .ThenInclude(ta => ta.Track)
+                        .ThenInclude(t => t.Genre)
+                .Include(a => a.Albums)
+                    .ThenInclude(al => al.AlbumType)
+                .FirstOrDefaultAsync(a => a.ArtistId == id);
+
+            return artist;
         }
 
-        public async Task<Artist> GetById(int id)
+        public async Task<IEnumerable<Artist>> GetAll()
         {
-            var artist = await _databaseContext.Artists.FindAsync(id);
-            return artist;
+            var artists = await _databaseContext.Artists
+                .Include(a => a.Image)
+                    .ThenInclude(i => i.ImageType)
+                .Include(a => a.TrackArtists)
+                    .ThenInclude(ta => ta.Track)
+                        .ThenInclude(t => t.Audio)
+                .Include(a => a.TrackArtists)
+                    .ThenInclude(ta => ta.Track)
+                        .ThenInclude(t => t.Genre)
+                .Include(a => a.Albums)
+                    .ThenInclude(al => al.AlbumType)
+                .ToListAsync();
+
+            return artists;
         }
 
         public async Task SaveChanges()
         {
             await _databaseContext.SaveChangesAsync();
         }
 
-        public async Task Update(Artist entity)
+        public async Task Create(Artist entity)
         {
-            var artist = await _databaseContext.Artists.FindAsync(entity.ArtistId);
+            await _databaseContext.AddAsync(entity);
+        }
 
-            artist.ArtistName = entity.ArtistName;
-            artist.ArtistBio = entity.ArtistBio;
-            artist.Image = entity.Image;
-            artist.ImageId = entity.ImageId;
+        public async Task Update(Artist entity)
+        {
+            // _databaseContext.Entry(entity).State = EntityState.Modified;
+            _databaseContext.Artists.Update(entity);
         }
 
-        public async void Delete(Artist entity) 
+        public void Delete(Artist entity) 
         {
-            throw new NotImplementedException(); //
+            _databaseContext.Remove(entity);
+            _databaseContext.SaveChanges();
```

Comment:
You have **SaveChanges** method in your repository. Other methods do not call DbContext SaveChanges except this. Remove this SaveChanges call here, or you will be confused in the future